From 027b97e4075f20bfcf4151c33f34b03bab7bdf32 Mon Sep 17 00:00:00 2001
From: Edsko de Vries <edsko@well-typed.com>
Date: Fri, 22 Feb 2013 14:07:07 +0000
Subject: [PATCH 2/2] Pass renamer results through typechecker

This is necessary because the type checker does not give us quite the full AST;
see http://www.haskell.org/pipermail/ghc-devs/2013-February/000540.html.
---
 compiler/main/HscMain.hs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/compiler/main/HscMain.hs b/compiler/main/HscMain.hs
index a899b25..00b1a18 100644
--- a/compiler/main/HscMain.hs
+++ b/compiler/main/HscMain.hs
@@ -848,7 +848,7 @@ hscFileFrontEnd :: ModSummary -> Hsc TcGblEnv
 hscFileFrontEnd mod_summary = do
     hpm <- hscParse' mod_summary
     hsc_env <- getHscEnv
-    tcg_env <- tcRnModule' hsc_env mod_summary False hpm
+    tcg_env <- tcRnModule' hsc_env mod_summary True hpm
     return tcg_env
 
 --------------------------------------------------------------
-- 
1.7.12.4 (Apple Git-37)

